### PR TITLE
Laboratory work #1, Anna Egorova - 21FPL1

### PR DESCRIPTION
**Pull request by chezar0**

### CODE REVIEW

File: lab_1_keywords_tfidf/main.py [hunk 1]
```diff
@@ -66,7 +93,20 @@ def get_top_n(frequencies: dict[str, Union[int, float]], top: int) -> Optional[l
...
+    if not isinstance(frequencies, dict) or not isinstance(top, int) or isinstance(top, bool) or frequencies == {}:
+        return None
+    if top <= 0:
+        return None
+    for i in frequencies.items():
+        if not isinstance(i[0], str) or not isinstance(i[1], (int, float)):
+            return None
+    sorted_list = []
+    sort = sorted(frequencies.items(), reverse=True, key=lambda x: x[1])
+    for i in sort:
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:105]: Why do you do this?

File: lab_1_keywords_tfidf/main.py [hunk 2]
```diff
@@ -66,7 +93,20 @@ def get_top_n(frequencies: dict[str, Union[int, float]], top: int) -> Optional[l
...
+        sorted_list.append(i[0])
+    if top > len(frequencies):
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:107]: It is not needed

File: lab_1_keywords_tfidf/main.py [hunk 3]
```diff
@@ -99,7 +148,22 @@ def calculate_tfidf(term_freq: dict[str, float], idf: dict[str, float]) -> Optio
...
+            return None
+    for i in idf.items():
+        if not isinstance(i[0], str) or not isinstance(i[1], float):
+            return None
+    dictionary = {}
+    for key in term_freq:
+        if key in idf:
+            tfidf = term_freq[key] * idf[key]
+        else:
+            tfidf = term_freq[key] * math.log(47 / (0 + 1))
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:164]: Why `(0 + 1)` ?

File: lab_1_keywords_tfidf/main.py (hunk 1 above)
Comment (chezar0) [reply lab_1_keywords_tfidf/main.py:105]: if you're talking about line 104, i wanted to get a list of values from frequencies in a descending order

File: lab_1_keywords_tfidf/main.py (hunk 3 above)
Comment (chezar0) [reply lab_1_keywords_tfidf/main.py]: thought i should write this in detail, fixed

File: lab_1_keywords_tfidf/main.py [hunk 4]
```diff
@@ -15,9 +17,14 @@ def clean_and_tokenize(text: str) -> Optional[list[str]]:
     Returns:
     list[str]: A sequence of lowercase tokens with no punctuation
 
-    In case of corrupt input arguments, None is returned
+    In case of corrupt arguments, None is returned
     """
-    pass
+    if isinstance(text, str):
```
[the remark this answers is not in the document]
Comment (chezar0) [reply lab_1_keywords_tfidf/main.py]: done

File: lab_1_keywords_tfidf/main.py (hunk 2 above)
Comment (chezar0) [reply lab_1_keywords_tfidf/main.py]: done

**Pull request closed without merging**